test/vstest.console.UnitTests/InProcessVsTestConsoleWrapperTests.cs

```diff
@@ -0,0 +1,829 @@
+        _mockTestRequestManager.Setup(trm => trm.ResetOptions()).Callback(() => { });
+
+        _mockOutput = new Mock<IOutput>();
+        _executor = new Executor(_mockOutput.Object, new Mock<ITestPlatformEventSource>().Object, new ProcessHelper(), new PlatformEnvironment());
+        _mockEventSource = new Mock<ITestPlatformEventSource>();
+
+        _consoleWrapper = new InProcessVsTestConsoleWrapper(
+            new ConsoleParameters(),
+            _mockRequestSender.Object,
+            _mockTestRequestManager.Object,
+            _executor,
+            _mockEventSource.Object);
+    }
+
+    [TestMethod]
+    public void InProcessWrapperConstructorShouldThrowIfPortIsInvalid()
+    {
+        _mockRequestSender.Setup(rs => rs.InitializeCommunication()).Returns(-1);
+
+        Assert.ThrowsException<TransationLayerException>(() =>
+            new InProcessVsTestConsoleWrapper(
+                new ConsoleParameters(),
+                _mockRequestSender.Object,
+                _mockTestRequestManager.Object,
+                new Executor(_mockOutput.Object, new Mock<ITestPlatformEventSource>().Object, new ProcessHelper(), new PlatformEnvironment()),
+                new Mock<ITestPlatformEventSource>().Object));
+    }
+
+    [TestMethod]
+    public void InProcessWrapperConstructorShouldSetEnvironmentVariablesReceivedAsConsoleParameters()
+    {
+        const string environmentVariableName = "AAAAA";
+
+        Environment.GetEnvironmentVariable(environmentVariableName).Should().BeNull();
+
+        var consoleParams = new ConsoleParameters();
+        consoleParams.EnvironmentVariables.Add(environmentVariableName, "1");
+
+        var _ = new InProcessVsTestConsoleWrapper(
+            consoleParams,
+            _mockRequestSender.Object,
+            _mockTestRequestManager.Object,
+            new Executor(_mockOutput.Object, new Mock<ITestPlatformEventSource>().Object, new ProcessHelper(), new PlatformEnvironment()),
+            new Mock<ITestPlatformEventSource>().Object);
+
+        Environment.GetEnvironmentVariable(environmentVariableName).Should().NotBeNull();
+    }
+
+    [TestMethod]
+    public void InProcessWrapperDiscoverTestsWithThreeParamsIsSuccessfullyInvoked()
+    {
+        var discoveryEventHandler = new Mock<ITestDiscoveryEventsHandler>();
+
+        DiscoveryRequestPayload? payload = null;
+        _mockTestRequestManager
+            .Setup(trm => trm.DiscoverTests(
+                It.IsAny<DiscoveryRequestPayload>(),
+                It.IsAny<ITestDiscoveryEventsRegistrar>(),
+                It.IsAny<ProtocolConfig>()))
+            .Callback((
+                DiscoveryRequestPayload p,
+                ITestDiscoveryEventsRegistrar _,
+                ProtocolConfig _) => payload = p);
+
+        _consoleWrapper.DiscoverTests(_testSources, _runSettings, discoveryEventHandler.Object);
+
+        Assert.IsTrue(_testSources.SequenceEqual(payload!.Sources!));
```

Comment:
[optional] You could do `Assert.IsNotNull(payload)` so you can remove all these bangs.

Reply:
You can do it for all the tests I think :)

Reply:
Fixed, thanks.